Python/gc_free_threading.c

```diff
@@ -1704,8 +1709,13 @@ PyObject_GC_Del(void *op)
     }
 
     record_deallocation(_PyThreadState_GET());
-
-    PyObject_Free(((char *)op)-presize);
+    PyObject *self = (PyObject *)op;
+    if (_PyObject_GC_IS_SHARED(self)) {
+        _PyObject_FreeDelayed(((char *)op)-presize);
```

Comment:
Possibly untrack the object here:

```suggestion
        _PyObject_GC_UNTRACK(op);
        _PyObject_FreeDelayed(((char *)op)-presize);
```